Source/WebKit/UIProcess/ProcessThrottler.cpp

```diff
@@ -336,11 +340,23 @@ void ProcessThrottler::setShouldTakeSuspendedAssertion(bool shouldTakeSuspendedA
     m_shouldTakeSuspendedAssertion = shouldTakeSuspendedAssertion;
 }
 
-void ProcessThrottler::delaySuspension()
+PageAllowedToRunInTheBackgroundCounter::Token ProcessThrottler::pageAllowedToRunInTheBackgroundToken()
+{
+    return m_pageAllowedToRunInTheBackgroundCounter.count();
+}
+
+void ProcessThrottler::numberOfPagesAllowedToRunInTheBackgroundChanged()
 {
-    PROCESSTHROTTLER_RELEASE_LOG("delaySuspension");
+    if (m_pageAllowedToRunInTheBackgroundCounter.value())
+        return;
```

Comment:
We delete the assertion when `m_pageAllowedToRunInTheBackgroundCounter` goes from 1 => 0 but we don't create one when it goes from 0 => 1. So I guess this RefCounter is only really meant to suppress dropping the assertion and `setThrottleState` is still in charge of creating an assertion if necessary?

Reply:
The idea is that if you're able to go from 0 to 1, then you were not suspended. If you weren't suspended then we were holding a process assertion already.